### DISCUSSION TITLE
Curious about Prerendering When Adding getServerSideProps

### DISCUSSION THREAD

**Discussion by uknowsj** [vercel/next.js#64012] [Help]
### Summary

When statically rendering, I know that useRouter.query is undefined because we can't access query information. 
However, when I added getServerSideProps, the query started receiving values, even though I didn't return anything as props. 

I'm not sure of the exact reason, but can someone explain?

### Additional information

```js
export const getServerSideProps: GetServerSideProps = async (context) => {
	return {
		props: {},
	};
};
export default function SomePage() {
	const router = useRouter(); //useRouter has the correct query from the first render
	const { id, data } = router.query;
	console.log('id', id, 'data', data);
...
}
```

**Comment by Gumaan-Mahar** [comment]
In Next.js, when you use `getServerSideProps`, it pre-renders the page on each request, allowing you to fetch data and pass it as props to your page component. This means that when you use `getServerSideProps`, the page is rendered on the server, and the data is fetched before sending the HTML to the client.

In your case, even though you're not explicitly returning any props from `getServerSideProps`, the fact that you're using it means that the page is rendered on the server for each request. During this server-side rendering process, `useRouter().query` is populated with the query parameters of the incoming request.

Here's a breakdown of what's happening:

**1.** Client requests the page.
**2.** Next.js executes `getServerSideProps` on the server, populating `useRouter().query` with query parameters.
**3.** The page component **(SomePage)** is rendered on the server, utilizing the query parameters.
**4.** Pre-rendered HTML is sent to the client.
**5.** On the client side, **SomePage** component mounts again. At this point, `useRouter().query` is undefined as static rendering lacks access to query parameters. However, as the content is already pre-rendered, it doesn't affect functionality.
